models/appModel.js

```diff
@@ -58,12 +58,42 @@ exports.fetchArticleById = (article_id) => {
 
 exports.fetchArticleComments = (article_id) => {
     return db
-    .query(`SELECT * FROM comments WHERE article_id = $1;`, [article_id])
-    .then(({ rows: comments  }) => {
-        if (comments.length === 0) {
-            return exports.fetchArticleById(article_id) 
-            .then(() => [])
-        } else 
-        return comments 
-    })
-}
\ No newline at end of file
+        .query(`SELECT * FROM comments WHERE article_id = $1;`, [article_id])
+        .then(({ rows: comments }) => {
+            if (comments.length === 0) {
+                return exports.fetchArticleById(article_id)
+                    .then(() => [])
+            } else
+                return comments
+        })
+}
+
+exports.addComment = (article_id, username, comment, date) => {
+    return db
+        .query(`SELECT * from articles WHERE article_id = $1`, [article_id])
+        .then(({ rows: articles }) => {
+                if (articles.length === 0) {
+                return Promise.reject({
+                    status: 404,
+                    msg: "Not found"
+                })
+            } else return
+        })
+        .then(() => {
+            return db
+            .query(`SELECT * from users WHERE username = $1;`, [username])
+            .then(({ rows :users}) => { 
+                if (users.length === 0) {
+                    return Promise.reject({
+                        status: 401,
+                        msg: "Unauthorized"
+                    })
+                }
+                })
```

Comment:
This should be handled by an SQL error so throwing a custom error may not be necessary. See if you can lose this and handle it using the SQL error code 🙂